### ISSUE TITLE
Support for view (xaml) and view models being in different projects

### ISSUE THREAD

**Issue by rosneru** [FantasticFiasco/mvvm-dialogs-integrated-into-mvvm-light-toolkit#7]
In the solution I'm working on are two *.csproj* files: One project `UI.Desktop.WPF` which contains files like `App.xaml`, `App.xaml.cs`, `MainWindow.xaml`, `MainWindow.xaml.cs`, etc. and another project `Logic.UI` which contains the view models, only *.cs* files.

I'm trying to replicate your [mvvm-dialogs-integrated-into-mvvm-light-toolkit](https://github.com/FantasticFiasco/mvvm-dialogs-integrated-into-mvvm-light-toolkit) example app in this situation.

My `App.xaml.cs` looks like this:

    public partial class App : Application
    {
      public IServiceProvider Services { get; }

      public App()
      {
        Services = ConfigureServices();
      }
  
      protected override void OnStartup(StartupEventArgs e)
      {
        base.OnStartup(e);
  
        MainWindow app = new MainWindow();
  
        var mainViewModel = Services.GetRequiredService<MainViewModel>();
        app.DataContext = mainViewModel;
        app.Show();
      }
    
  
  
      private static IServiceProvider ConfigureServices()
      {
        // Services
        var services = new ServiceCollection();
        services.AddSingleton<IDialogService, DialogService>().AddTransient<MainViewModel>();
  
        // ViewModels
        //services.AddTransient<MessageBoxWindowViewModel>();
  
        return services.BuildServiceProvider();
      }
    }

And the basic dialog `dialogService.ShowMessageBox(...)` already works with it.

But I wonder how to use a *CustomDialogTypeLocator* as you explained [there](https://github.com/FantasticFiasco/mvvm-dialogs/wiki/Custom-dialog-type-locators) in this situation.

How could I extend my `App.xaml.cs` to use the Custom dialog type locator? By something like you described in your example:

    dialogService = new DialogService(dialogTypeLocator: new MyCustomDialogTypeLocator());

So currently I'm not using the dialog type locator and of course when the view model tries to open the dialog like:

   var openDialog = new OpenDeviceDialogViewModel();
   var success = dialogService.ShowDialog(this, openDialog);

it gives a run time error:

*System.TypeLoadException: 'Dialog with name 'Logic.UI.DialogViewModels.OpenDeviceDialog' is missing.*

*If your project structure doesn't conform to the default convention of MVVM Dialogs you can always define a new convention by implementing your own dialog type locator. For more information on how to do that, please read the GitHub wiki or ask the author.'*

And thats right, because in my case is the dialog is situated the view project, in `UI.Desktop.WPF.Dialogs.OpenDeviceDialog`. So I need the locator somehow.

**Comment by FantasticFiasco** [comment]
hi @rosneru! Would it be possible for you to fork this repo and then update it to your structure, and then I'm sure we'll figure this one out.

**Comment by rosneru** [comment]
Hi @FantasticFiasco, thanks for your reply!

I think forking the repository isn't necessary as I found a solution already: I modified the DialogTypeLocator to fit to my project structure and now it works.

I instantiate it in `App.xaml.cs::OnStartup()`  method like this:

      var dialogService = new DialogService(dialogTypeLocator: new DialogTypeLocator());
      var mainViewModel = new MainViewModel(dialogService);

My only remaining question would be how to change that instantiation to something like:

    services.AddSingleton<IDialogService, DialogService>().AddTransient<MainViewModel>();

and passing my custom DialogTypeLocator to it. But I guess, this is not a topic related to your library. I think reading [that docs](https://docs.microsoft.com/en-us/windows/communitytoolkit/mvvm/ioc?source=recommendations) is my next task to do would bring a solution.

So again, thanks for your replay and many thanks for creating this library!
